File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] foundTileSize(int totalTile, int brown){
        for(int i=1; i < Math.sqrt(totalTile)+1; i++){
            if(totalTile % i == 0){
                int j = totalTile / i;
                int countBrown = i*2 + j*2 -4;
                if(brown == countBrown){
                    int[] arr = {j, i};
                    return arr;
                }
            }
        }
                    
        int[] arr = {0,0};
        return arr;
    }
    
    public int[] solution(int brown, int yellow) {
        int totalTile = brown + yellow;
        int[] answer = foundTileSize(totalTile, brown);
        
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
import math

def solution(brown, yellow):
    answer = []
    total = brown + yellow
    
    for row in range(1, int(math.sqrt(total))+1):
        if total % row == 0:
            column = total // row
            yellow_count = (column -2) * (row -2)
            if yellow_count == yellow:
                answer.append(column)
                answer.append(row)
            else:
                continue
    
    return answer



# brown + yellow = 전체 타일의 개수
# n x m = 전체 타일, 완전 탐색
# 항상 가로길이 >= 세로 
# 4*3 = 전체 타일 (4-2) * (3-1) = yellow
